src/component/routers/GroupsRouter.tsx

```diff
@@ -27,19 +27,20 @@ export type GroupsRouterProps = GroupsRouterStoreProps & GroupsRouterPassedProps
 
 export const extractGroups = ({ exits, router }: Node): SearchResult[] =>
     (router as SwitchRouter).cases.map(kase => {
-        const resultName = exits.reduce((result, { name, uuid }) => {
+        let resultName = '';
+        for (const { name, uuid } of exits) {
             if (uuid === kase.exit_uuid) {
-                result += name;
+                resultName += name;
+                break;
             }
-            return result;
-        }, '');
+        }
         return { name: resultName, id: kase.arguments[0] };
     });
 
-export const groupSplitExistsAtNode = (node: Node) =>
+export const hasGroupsRouter = (node: Node) =>
     hasSwitchRouter(node) && hasWait(node, WaitType.group);
 
-export class GroupsRouter extends React.PureComponent<GroupsRouterProps> {
+export class GroupsRouter extends React.Component<GroupsRouterProps> {
```

Comment:
no sense making this a `PureComponent` instance since it's being wrapped by `connect`